### DISCUSSION TITLE
Can I use this library with query string instead of subdirectories?

### DISCUSSION THREAD

**Discussion by mariusbolik** [Alexandre-Fernandez/astro-i18n#79] [Q&A]
Hello,

thank you very much for this library!
I have just a short question: I don't really need SEO advantages, so is it possible to use this library with query strings e.g `?lang=en`, `?lang=cn` instead of subdirectories like `/en`, `/cn`?
So I don't need to run `i18n:sync` and generate all the files.

Kind regards
Marius

**Comment by Alexandre-Fernandez** [comment]
Hi, at the moment this feature is not implemented. Is running `i18n:sync` the only reason why you need this ?

A workaround that could probably work is adding your own middleware just after the astroI18n one where you would set the route to what it should be for example from the query string : 
```ts
astroI18n.route = `/${ctx.url.searchParams.get("lang")}${ctx.url.pathname}`
```

I'm looking forward to add a way to automatically run the sync command in the future.

**Reply by mariusbolik** [reply]
Thank you for the quick response. Mainly, I don't want to have all the subfolders in my project and in the url.
I use Astro for Security and Performance Reasons, not for SEO. So it would be great if a user can visit `/account` in different languages, without `/en`, `/fr` in the URL and without all the generated folders in my file system. Astro should replace the text placeholders directly using SSR. Is this possible? :)

**Reply by mariusbolik** [reply]
And I am always getting an error. Do you have any idea on that?
<img width="1004" alt="Bildschirmfoto 2024-02-08 um 16 16 33" src="https://github.com/Alexandre-Fernandez/astro-i18n/assets/39862171/38fb6839-5c36-4076-bd63-42c985ffd092">
